src/zui/ZUIRatingChart/index.tsx

```diff
@@ -0,0 +1,219 @@
+}
+
+const useStyles = makeStyles((theme) => {
+  return {
+    root: {
+      '& .average .averageRating': {
+        fill: 'var(--dataMain)',
+        fontSize: 13,
+        transition: 'opacity 0.25s',
+      },
+      '& .average .averageRatingBg': {
+        fill: '#FFFFFF',
+        filter: 'blur(5px)',
+        height: 32,
+        width: 50,
+      },
+      '& .chart': {
+        fill: 'var(--dataFaded)',
+      },
+      '& .data .dot': {
+        opacity: 0,
+      },
+      '& .dot': {
+        fill: 'var(--dataMain)',
+        transitionAttribute: 'fill, opacity',
+        transitionDuration: '.25s',
+      },
+      '& .indexNumber': {
+        fill: theme.palette.secondary.light,
+        fontSize: 13,
+        textAnchor: 'middle',
+      },
+      '& .ratingCount': {
+        fill: 'var(--dataMain)',
+        fontSize: 13,
+        opacity: 0,
+        textAnchor: 'middle',
+        transition: 'opacity .25s',
+      },
+      '& .track': {
+        fill: 'var(--dataFocus)',
+        transition: 'fill .25s',
+      },
+      '& svg': {
+        '--dataFaded': lighten(theme.palette.primary.main, 0.83),
+        '--dataFocus': lighten(theme.palette.primary.main, 0.63),
+        '--dataMain': theme.palette.primary.main,
+        cursor: 'pointer',
+        overflow: 'visible',
+      },
+      '&:hover .average .averageRating': {
+        opacity: 0,
+      },
+      '&:hover .average .dot': {
+        fill: 'var(--dataFocus)',
+      },
+      '&:hover .chart': {
+        fill: 'var(--dataFocus)',
+      },
+      '&:hover .data .dot': {
+        opacity: 1,
+      },
+      '&:hover .ratingCount': {
+        opacity: 1,
+      },
+      '&:hover .track': {
+        fill: 'var(--dataFaded)',
+      },
+    },
+  };
+});
+
+const ZUIRatingChart: FC<ZUIRatingChartProps> = ({
+  description,
+  title,
+  visualizationHeight = 50,
+  data,
+}) => {
+  const classes = useStyles();
+
+  let ratingTotals = 0;
+  let numRatings = 0;
+  if (data.every((val) => val >= 0)) {
+    data.forEach((val, i) => {
+      numRatings += val;
+      ratingTotals += val * (i + 1);
+    });
+  } else {
+    return null;
+  }
+
+  const avg = ratingTotals / numRatings;
+  const maxRating = 5;
+  const avgPos = (avg - 1) / (maxRating - 1);
+
+  return (
+    <Box className={classes.root} p={2}>
+      {title && <Typography pb={1}>{title}</Typography>}
+      {description && (
+        <Typography color="secondary" pb={2}>
+          {description}
+        </Typography>
+      )}
+      <ZUIResponsiveContainer ssrWidth={200}>
+        {(width: number) => {
+          const baseNumber = 4;
+          const highestValue = Math.max(...data, 1);
+          let path = `M ${baseNumber} ${visualizationHeight}`;
+          const dots: { x: number; y: number }[] = [];
+          const graphWidth = width - baseNumber * 2;
+
+          path += data
+            .map((val, i) => {
+              const x = i * (graphWidth / (data.length - 1 || 1)) + baseNumber;
```

Comment:
It looks to me like `(graphWidth / (data.length - 1 || 1))` is a recurring expression. It would be useful to put it in a variable and give it a name that makes it more obvious what that expression means (e.g. `widthPerSegment`).